packages/engine/src/scene/classes/Clouds.ts

```diff
@@ -0,0 +1,210 @@
+    const planeGeometry = new PlaneBufferGeometry(1, 1, 1, 1)
+    const geometry = new InstancedBufferGeometry()
+    geometry.index = planeGeometry.index
+    geometry.attributes = planeGeometry.attributes
+
+    const particleItemSize = 4
+    const particleArray = []
+    const zRotationArray = []
+
+    for (let x = 0; x < this.dimensions.x; x++) {
+      for (let y = 0; y < this.dimensions.y; y++) {
+        for (let z = 0; z < this.dimensions.z; z++) {
+          particleArray.push(
+            this.getNoise3D(
+              x / this.noiseZoom.x + this.noiseOffset.x,
+              y / this.noiseZoom.x + this.noiseOffset.x,
+              z / this.noiseZoom.x + this.noiseOffset.x
+            ) *
+              this.worldScale.x -
+              this.worldScale.x * 0.5
+          ) // X
+          particleArray.push(
+            this.getNoise3D(
+              x / this.noiseZoom.y + this.noiseOffset.y,
+              y / this.noiseZoom.y + this.noiseOffset.y,
+              z / this.noiseZoom.y + this.noiseOffset.y
+            ) *
+              this.worldScale.y -
+              this.worldScale.y * 0.5
+          ) // Y
+          particleArray.push(
+            this.getNoise3D(
+              x / this.noiseZoom.z + this.noiseOffset.z,
+              y / this.noiseZoom.z + this.noiseOffset.z,
+              z / this.noiseZoom.z + this.noiseOffset.z
+            ) *
+              this.worldScale.z -
+              this.worldScale.z * 0.5
+          ) // Z
+
+          particleArray.push(
+            this.getNoise3D(x / 7 + 2777, y / 7 + 2777, z / 7 + 2777) * this.spriteScaleRange.y +
+              this.spriteScaleRange.x
+          ) // Scale
+
+          zRotationArray.push(this.getNoise3D(x / 3, y / 3, z / 3) * Math.PI)
+        }
+      }
+    }
+
+    geometry.setAttribute('particles', new InstancedBufferAttribute(new Float32Array(particleArray), particleItemSize))
+    geometry.setAttribute('particleAngle', new InstancedBufferAttribute(new Float32Array(zRotationArray), 1))
+    this.geometry = geometry
+    ;(this.material as any).uniforms.fogColor.value = this.fogColor
+    ;(this.material as any).uniforms.fogNear.value = this.fogRange.x
+    ;(this.material as any).uniforms.fogFar.value = this.fogRange.y
+  }
+
+  update(dt: number) {}
+
+  copy(source: this, recursive = true) {
+    super.copy(source, recursive)
+
+    const material = (this as any).material as RawShaderMaterial
+    const sourceMaterial = (source as any).material as RawShaderMaterial
+
+    material.uniforms.map.value = sourceMaterial.uniforms.map.value
+
+    this.worldScale.copy(source.worldScale)
+    this.dimensions.copy(source.dimensions)
+    this.noiseZoom.copy(source.noiseZoom)
+    this.noiseOffset.copy(source.noiseOffset)
+    this.spriteScaleRange.copy(source.spriteScaleRange)
+    this.fogColor.copy(source.fogColor)
+    this.fogRange.copy(source.fogRange)
+
+    return this
+  }
+
+  static fromArgs(args): Promise<Clouds> {
```

Comment:
please use the AssetLoader class for loading textures

Reply:
Removed the function since it was not used